File: test_resources/rust/enum_models.rs
#[derive(Debug, Clone, PartialEq)]
pub enum MyEnum {
    Idle,
    Move {
        x: f64,
        y: f64,
    },
    Update(
        f64,
        f64,
        String,
    ),
}

impl Default for MyEnum {
    fn default() -> Self {
        Self::Idle
    }
}

#[derive(Debug, Clone, PartialEq)]
pub enum MyEnumWithoutPositions {
    Option1,
    Option2,
    Option3,
    Option4,
}

impl Default for MyEnumWithoutPositions {
    fn default() -> Self {
        Self::Option1
    }
}

File: test_resources/swift/enum_models.swift
enum MyEnum {
    case idle
    case move(
        /* x */ Double,
        /* y */ Double
    )
    case update(
        Double,
        Double,
        String
    )

    static func createDefault() -> MyEnum {
        return .idle
    }
}

enum MyEnumWithoutPositions {
    case option1(
        UInt64
    )
    case option2(
        /* name */ String
    )
    case option3
    case option4

    static func createDefault() -> MyEnumWithoutPositions {
        return .option1(
            0
        )
    }
}

enum MyEnumWithManyArguments {
    case option1(
        UInt64,
        UInt64,
        UInt64
    )
    case option2(
        /* name */ String
    )
    case option3
    case option4

    static func createDefault() -> MyEnumWithManyArguments {
        return .option1(
            0,
            0,
            0
        )
    }
}

enum MyEnumWithNamedArguments {
    case option1(
        /* name */ String,
        /* value */ Float,
        /* bytes */ [UInt8]
    )
    case option2(
        UInt64,
        UInt64,
        UInt64
    )
    case option3
    case option4

    static func createDefault() -> MyEnumWithNamedArguments {
        return .option1(
            /* name */ "",
            /* value */ 0,
            /* bytes */ []
        )
    }
}
